sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-redis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/redis/JedisRedisCheckpointStore.java

```diff
@@ -74,35 +88,39 @@ public Mono<Void> updateCheckpoint(Checkpoint checkpoint) {
                 .propagate(new IllegalStateException(
                     "Checkpoint is either null, or both the offset and the sequence number are null.")));
         }
-
+        String prefix = prefixBuilder(checkpoint.getFullyQualifiedNamespace(), checkpoint.getEventHubName(), checkpoint.getConsumerGroup());
         String key = keyBuilder(checkpoint.getFullyQualifiedNamespace(), checkpoint.getEventHubName(), checkpoint.getConsumerGroup(), checkpoint.getPartitionId());
-        Map<String, String> checkpointStorageMap = new HashMap<>();
+        JSONObject checkpointInformation = new JSONObject();
         try (Jedis jedis = jedisPool.getResource()) {
-            //Here is where you add everything to the checkpointStorageMap
+            checkpointInformation.put("CHECKPOINT_NAME", key);
             String sequenceNumber = checkpoint.getSequenceNumber() == null ? null
                 : String.valueOf(checkpoint.getSequenceNumber());
-            checkpointStorageMap.put("SEQUENCE_NUMBER", sequenceNumber);
-
+            checkpointInformation.put("SEQUENCE_NUMBER", sequenceNumber);
             String offset = checkpoint.getOffset() == null ? null
                 : String.valueOf(checkpoint.getOffset());
-            checkpointStorageMap.put("OFFSET", offset);
-            jedis.hmset(key, checkpointStorageMap); // key -> { field: value, field:value}
+            checkpointInformation.put("OFFSET", offset);
+            //Before adding this checkpoint to the list of Checkpoints, check if it is the most updated version of the checkpoint.
+            jedis.sadd(prefix, checkpointInformation.toString());
             jedisPool.returnResource(jedis);
         }
         return null;
     }
+
+    private String prefixBuilder(String fullyQualifiedNamespace, String eventHubName, String consumerGroup) {
+        return fullyQualifiedNamespace + "/" + eventHubName + "/" + consumerGroup;
+    }
     private String keyBuilder(String fullyQualifiedNamespace, String eventHubName,  String consumerGroup, String partitionId) {
         return fullyQualifiedNamespace + "/" + eventHubName + "/" + consumerGroup + "/" + partitionId;
     }
     private Boolean isCheckpointValid(Checkpoint checkpoint) {
         return !(checkpoint == null || (checkpoint.getOffset() == null && checkpoint.getSequenceNumber() == null));
     }
-    private JedisPoolConfig createPoolConfig(RedisClientConfig config) {
-        JedisPoolConfig poolConfig = new JedisPoolConfig();
-        poolConfig.setMaxTotal(config.getPoolMaxTotal());
-        poolConfig.setMaxIdle(config.getPoolMaxIdle());
-        poolConfig.setBlockWhenExhausted(config.getPoolBlockWhenExhausted());
-        poolConfig.setMinIdle(config.getPoolMinIdle());
-        return poolConfig;
-    }
+//    private JedisPoolConfig createPoolConfig(RedisClientConfig config) {
```

Comment:
you can delete this code. the power of git is you can check the previous version of this file if you need the contents back